source/simulator/assembly.cc

```diff
@@ -1419,7 +1419,8 @@ namespace aspect
                    * on the direction of the flow. Thus the flow-dependent terms below are
                    * only calculated if the edge is an inflow edge.
                    */
-                  const bool inflow = ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.);
+                  const bool inflow = (advection_field.is_temperature()
+                                       || ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.));
```

Comment:
Can you explain this change in more detail? It makes all boundaries inflow boundaries if we are considering the temperature. Maybe that's what we want, but I don't understand why this would be so.